### DISCUSSION TITLE
docs: add text descriptions to API reference

### DISCUSSION THREAD

**Discussion by monitz87** [mikro-orm/mikro-orm#4257] [General]
First of all, kudos on the library overall and the docs in general. So far I've been very satisfied with MikroORM.

That said, I come with a suggestion on how to greatly improve the dev experience.

Despite the API reference docs being very thorough in terms of available classes with their properties and methods (and their parameters), there's a severe lack of text descriptions.

From what I've seen, almost no methods explain their behavior, caveats, different branches and possible return values. Parameters, their possible values and what they do aren't explained either.

When trying to do more advanced stuff (for instance, trying to work directly with `UnitOfWork` inside an `onFlush` call), it becomes necessary to be familiar with the internals of the library, and things like adequate, descriptive documentation would do wonders when you're not sure why what you're trying to do isn't working as intended.

**Comment by B4nan** [comment]
It's all about priorities, I would love to do that, but I have no time for it, not right now and it won't be better in foreseeable future. I will be more than happy to accept PRs for this. 

The priority is: fixing bugs > finishing v6 > finishing getting started guide > updating regular docs > jsdocs.

**Reply by monitz87** [reply]
I would love to submit a PR for this but I can't do it if I don't know what the methods do. It's a chicken and egg sort of deal :(

**Reply by B4nan** [reply]
I can handle the UoW methods myself, that is indeed a bit trickier to grasp, but the vast majority of public methods that don't have JSDoc should be rather easy to understand, and I would like to see some help with those. I don't think this is chicken-egg problem, its more like "I have too much stuff to do and I need help" problem :] In the end, there are regular docs, issues, _many_ integration tests, and the code itself, so you can always dive deeper and find the answers even for the UoW stuff.

**Reply by monitz87** [reply]
I'll try to find time to document some of the more obvious classes then, but if I do submit a PR with good enough documentation, I ask that you reconsider assigning a higher priority to the UoW docs. This is not a condition, but simply a request.

**Reply by B4nan** [reply]
I definitely want to improve the docs in upcoming weeks, before v6 gets out. One of the things I want to cover is internals, so mainly change tracking and the architecture behind it. The priority is still docs before jsdocs, but I will try to improve that too.
